scRNA-seq/post-processing_scRNA-seq.Rmd

```diff
@@ -0,0 +1,150 @@
+---
+title: "scRNA-seq_post_processing"
+output: html_notebook
+---
+
+**CCDL 2019**
+
+In this notebook, we'll perform quality control analyses and normalization of 
+scRNA-seq count data. 
+
+As opposed to bulk rna-seq, there are there are a few main things to look out for
+in single cell rna-seq:
+
+**Single-cell rna-seq has...**  
+- More amplification (and more of it's associated biases and error)  
+- More 0's in the gene expression data (most genes aren't expressed across cell types)  
+
+This script is adapted from [Seurat pbmc tutorial](https://satijalab.org/seurat/pbmc3k_tutorial.html) 
+and from [COMBINE lab's alevin tutorial](https://combine-lab.github.io/alevin-tutorial/2018/alevin-seurat/)
+
+For these analyses, we will need Seurat and dplyr packages. 
+```{r Install Packages}
+# Attach library
+library(Seurat)
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+```
+
+#### Perform QC checks with 'csoneson/alevinQC'
+This will provide html output with graphs evaluating the data. Keep in mind that 
+the data we have processed in this workshop is not the full dataset, and won't 
+look as good as the full set.
+```{r AlevinQC }
+# Store the path to the alevin output files
+alevin.file <- file.path("data", "alevin_output", "pbmc_1k_v2_S1_L001_subset")
+
+# Produce a QC report
+alevinQC::alevinQCReport(alevin.file,
+                         sampleId = "pbmc_1k_v2_S1_L001_subset", 
+                         outputFile = "pbmc_1k_v2_S1_L001_qc_report.html", 
+                         outputFormat = "html_document")
+```
+
+#### Import counts from alevin output
+This chunk of code is from the [COMBINE lab's tutorial](https://combine-lab.github.io/alevin-tutorial/2018/alevin-seurat/)
+```{r Import Data}
+# Set up function from the COMBINE lab
+ReadAlevin <- function(base.path = NULL){
+  if (! dir.exists(base.path )){
+    stop("Directory provided does not exist")
+  }
+  # Obtain paths to data
+  barcode.loc <- file.path(base.path, "alevin", "quants_mat_rows.txt")
+  gene.loc <- file.path(base.path, "alevin", "quants_mat_cols.txt")
+  matrix.loc <- file.path(base.path, "alevin", "quants_mat.csv")
+    
+  if (!file.exists(barcode.loc)) {
+    stop("Barcode file missing")
+  }
+  if (!file.exists(gene.loc)) {
+    stop("Gene name file missing")
+  }
+  if (!file.exists(matrix.loc)) {
+    stop("Expression matrix file missing")
+  }
+  # Read in the data from alevin output
+  matrix <- as.matrix(read.csv(matrix.loc, header=FALSE))
+  matrix <- t(matrix[,1:ncol(matrix)-1])
+    
+  # Read in the cell and gene names from the alevin output
+  cell.names <- readLines(barcode.loc)
+  gene.names <- readLines(gene.loc)
+    
+  # Apply the colnames and rownames to the dataset
+  colnames(matrix) <- cell.names
+  rownames(matrix) <- gene.names
+    
+  # Make NA values into 0's
+  matrix[is.na(matrix)] <- 0
+  return(matrix)
+}
+
+# Run this function on our files
+alv.data <- ReadAlevin(alevin.file)
```

Comment:
This was a _little_ bit slow on my MacBook, but it is a big matrix. Quick thoughts on possible improvements: change how the CSV files are read in (i.e., don't use base R), perhaps can make the transpose step more efficient (less sure about this)

Reply:
Done.

Reply:
It is much more efficient to use `readr::read_csv` (as we kinda knew already anyway).